src/deferred.rs

```diff
@@ -0,0 +1,140 @@
+use std::mem;
+use std::ptr;
+
+/// Number of words a piece of `Data` can hold.
+///
+/// Three words should be enough for the majority of cases. For example, you can fit inside it the
+/// function pointer together with a fat pointer representing an object that needs to be destroyed.
+const DATA_WORDS: usize = 3;
+
+/// Some space to keep a `FnOnce()` object on the stack.
+type Data = [usize; DATA_WORDS];
+
+/// A `FnOnce()` that is stored inline if small, or otherwise boxed on the heap.
+///
+/// This is a handy way of keeping an unsized `FnOnce()` within a sized structure.
+pub struct Deferred {
```

Comment:
There are two variants of `Deferred`: one using `data` as the function, and the other using `data` as a pointer to the function.  What about giving two variants to `enum Deferred`?